src/App.tsx

```diff
@@ -1,61 +1,159 @@
+import React from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
-
-export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/users" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
-    </div>
-  );
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+import users from './api/users';
+
+import { getUser } from './helpers/getUser';
+import { todos } from './helpers/getTodos';
+
+import { TodoList } from './components/TodoList';
+
+type State = {
+  currentTodo: Todo,
+  todos: Todo[],
+  inputStatusMonitoring: boolean,
 };
+
+export class App extends React.Component<{}, State> {
+  state = {
+    currentTodo: {
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+      title: '',
+      userId: 0,
+      completed: false,
+    },
+    todos: [...todos],
+    inputStatusMonitoring: false,
+  };
+
+  handleChange = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+  ) => {
+    const { name, value } = event.target;
+    const re = /^[A-Za-zА-Яа-я0-9 ]*$/i;
```

Comment:
**(JUST FYI)** I would move it into `src/constnats` and name it like `LETTERS_NUMBERS_REGEXP`